### ISSUE TITLE
Benchmark results

### ISSUE THREAD

**Issue by fmellomascarenhas** [OpenBioLink/OpenBioLink#64]
Hi,

I checked that you updated your benchmark table, which is very exciting. I have a few questions about it.

**1) TransR results difference:** Did you guys explain somewhere the difference between TransR (baseline) and TransR with dgl-KE? Is there any reason for the results with dgl-ke to be way better?

**2) Could the code please be provided?** The code links either don't work or they redirect to dgl-ke library;

**3) Could you please clarify if the prediction is about all the edges in the test dataset of HQ directed graph?** If so, then the prediction will be skewed towards edges that are more frequent. I wonder if there could be some sort of Micro/Macro F1 score, or a score per edge class. This paper shows the importance of doing it for Drug-Drug interaction: https://arxiv.org/pdf/2010.01450.pdf

![image](https://user-images.githubusercontent.com/23004953/100912872-07309180-349f-11eb-90a3-27e3cfa2db49.png)

Thanks!
Felipe

**Comment by matthias-samwald** [comment]
1) Dear Felipe, the new results (including TransR) were done after further hyperparameter search and with another library (baseline results were achieved with PyKEEN, newer results were achieved with DGL-KE). I can't really say which of the changes contributed more to the improved results, unfortunately. @LFGMUW -- Do you have any other remarks?

2) @LFGMUW Is there any additional code you wrote for the evaluation that we could share, or can the current evaluation files be used with the library as-is?

3) Yes, the main evaluation is done in a rather brute way across all relations, which makes frequent relations dominate the results. We also plan to provide per-relation results in future evaluations.

**Comment by fmellomascarenhas** [comment]
Thank you so much, @matthias-samwald. Just so you know, I am implementing a GNN with DGL using your dataset. I might have new questions along the way, but hopefully we will be able to contribute to this awesome work too, if my company allows me to. If you don't mind, I have one more question: 

We saw that a great percentage of the nodes in the dataset couldn't be mapped to other ontology systems (ids_no_mapping.csv). We wanted to try to solve this somehow, but it is hard, because we don't know how the data was aggregated. I haven't stopped to study your scripts yet, but do they show raw the raw data was processed/aggregated and IDs were mapped? [This post from AstraZeneca](https://towardsdatascience.com/modelling-biomedical-data-for-a-drug-discovery-knowledge-graph-a709be653168), for example, shows that one could use a disease parents ID to map children disease IDs, ex: Kidney disease > some specific kidney disease. Also, did you guys build some dictionary to map the ids? If so, maybe it would be an useful file to make available.

Thanks again!

**Comment by nomisto** [comment]
Hi @fmellomascarenhas ,

Unfortunately we do not have a centralized dictionary file for the mappings. The files used for mapping can be found [here](https://github.com/OpenBioLink/OpenBioLink/tree/master/src/openbiolink/graph_creation/metadata_db_file). The usages of the maps can be found in the [metadata_edges](https://github.com/OpenBioLink/OpenBioLink/tree/master/src/openbiolink/graph_creation/metadata_edge). The mapping files (infiles) of metadata_edges and metadata_db_files are connected via its respective reader and processor. Admittedly this is a little overwhelming at first glance. This is an overview of how these different classes are connected, that may help:

![overview](https://user-images.githubusercontent.com/28439912/102533617-f2aae680-40a5-11eb-8c7e-4303a9f1991f.PNG)

**Comment by LFGMUW** [comment]
> 
> 
>     1. Dear Felipe, the new results (including TransR) were done after further hyperparameter search and with another library (baseline results were achieved with PyKEEN, newer results were achieved with DGL-KE). I can't really say which of the changes contributed more to the improved results, unfortunately. @LFGMUW -- Do you have any other remarks?
> 
No, it seems also the training details for those frameworks of quite different kind, e.g. for the PyKEEN experiments, true negatives have been sourced and evaluation conditions differ.
>     2. @LFGMUW Is there any additional code you wrote for the evaluation that we could share, or can the current evaluation files be used with the library as-is?
> 
The evaluation methology has not been modified from the frameworks implementation so see that code for details

**Comment by fmellomascarenhas** [comment]
@nomisto, since I mentioned here the challenge regarding IDs without mapping, do you think that maybe this could be addressed in the [openbiolink 1.1](https://github.com/OpenBioLink/OpenBioLink/milestone/1), if you are going to do major changes in the code?

Maybe the [solution suggested by AstraZeneca](https://towardsdatascience.com/modelling-biomedical-data-for-a-drug-discovery-knowledge-graph-a709be653168) could be implemented, and nodes could have a secondary ID (their original, and the one mapped to a main ontology ID).

![image](https://user-images.githubusercontent.com/23004953/102543929-e5233b80-4068-11eb-9ead-b6f5b66aa6f1.png)

**Comment by nomisto** [comment]
That is very interesting! We will definitely consider it and see how we could apply this to OpenBioLink in a future release. However I have to clarify that the date this milestone is due is unclear at this point. As OpenBioLink aims to be a biomedical benchmark dataset, frequent bigger changes would be counterintuitive.

**Comment by matthias-samwald** [comment]
To clarify even further: We will only make some additions (node features representing e.g. molecular structure), but no breaking changes during 2021. Introducing changes too frequently would undermine the establishment of OpenBioLink as a benchmark. We will however collect all proposed improvements and release and updated version in early 2022.
